src/sql/parts/notebook/notebookActions.ts

```diff
@@ -253,44 +250,42 @@ export class KernelsDropdown extends SelectBox {
 	}
 
 	updateModel(model: INotebookModel): void {
-		this.model = model;
-		this._register(model.kernelsChanged((defaultKernel) => {
-			this.updateKernel(defaultKernel);
+		this.model = model as NotebookModel;
```

Comment:
Is this because you didn't change modelInterfaces to include the new events?

Reply:
As we looked code together, this._onProviderIdChanged.fire is called in constructor.